### PR TITLE
[13.0][ADD] account_move_exception & account_move_post_block

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Using https://github.com/OCA/server-tools/tree/13.0/base_exception.

Account Move Exception
---------------------------------

This module allows you attach several customizable exceptions to your account move in a way that you can filter moves by exceptions type and fix them.

This is especially useful in an scenario for mass accounts move import, because it's likely some moves have errors when you import them (like product not found in Odoo, wrong line format etc.)

Account Move Post Block
----------------------------------

This module allows you to block the post of a journal entry when a Post Block Reason has been provided. The moves will be waiting for post by a Manager.

### CODE REVIEW

File: account_move_exception/data/account_exception_data.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        <field name="name">No email on vendor</field>
+        <field name="description">No Email for Vendor</field>
+        <field name="sequence">50</field>
+        <field name="model">account.move</field>
+        <field name="code">if not self.partner_id.email:
+            failed=True</field>
+        <field name="active" eval="False" />
+    </record>
+    <record id="aml_excep_qty_check" model="exception.rule">
+        <field name="name">Quantity not negative</field>
```
Comment (HviorForgeFlow) [inline account_move_exception/data/account_exception_data.xml]: Change name to Quantity not positive or Quantity negative

**Review by HviorForgeFlow — APPROVED** [review]
Code review :+1: just a little comment

File: account_move_exception/data/account_exception_data.xml (hunk 1 above)
Comment (MiquelRForgeFlow) [reply account_move_exception/data/account_exception_data.xml]: Done!

**Review by chusamo — APPROVED** [review]
code review LGTM Thank you

**Review by AaronHForgeFlow — APPROVED** [review]

File: account_move_post_block/views/account_move_view.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        <field name="name">account.move.form.block.reason</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <button name="button_draft" position="attributes">
+                <attribute name="states">cancel</attribute>
+            </button>
+            <field name="date" position="after">
+                <field
+                    name="post_block_id"
```
Comment (JordiMForgeFlow) [inline account_move_post_block/views/account_move_view.xml]: We should make it invisible if post_block_id=False

File: account_move_exception/views/account_view.xml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            </header>
+            <xpath expr="//field[@name='date']/.." position="inside">
+                <field
+                    name="ignore_exception"
+                    states="account"
+                    groups='base_exception.group_exception_rule_manager'
+                />
+            </xpath>
+            <xpath expr="//field[@name='company_id']" position="after">
+                <field name="exception_ids" widget="many2many_tags" readonly="True" />
```
Comment (JordiMForgeFlow) [inline account_move_exception/views/account_view.xml]: Are exceptions dependent on the post_block_id field? Maybe the field could be hidden if no blocking is done.

**Comment by JordiMForgeFlow** [comment]
Should the _post block reason_ and _exceptions_ fields be added directly into the move, or would it be better to add a new group inside the Other Info tab or a new tab?

File: account_move_exception/views/account_view.xml (hunk 3 above)
Comment (MiquelRForgeFlow) [reply account_move_exception/views/account_view.xml]: no

Comment (MiquelRForgeFlow) [reply account_move_exception/views/account_view.xml]: @JordiMForgeFlow  could you check now if this solves your visibility issues?

**Comment by JordiMForgeFlow** [comment]
image

The exceptions are still always displayed. I think that Odoo has some trouble with many2many fields in views for these cases. Maybe it could be added into the Other Info tab?

**Comment by MiquelRForgeFlow** [comment]
> Maybe it could be added into the Other Info tab?

Done.

File: account_move_exception/views/account_view.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        parent="account.product_product_menu_sellable"
+        sequence="20"
+        name="Exceptions"
+    />
+    <menuitem
+        id="menu_account_exception_parent2"
+        parent="account.product_product_menu_purchasable"
+        sequence="20"
+        name="Exceptions"
+    />
```
Comment (JordiMForgeFlow) [inline account_move_exception/views/account_view.xml]: How about moving these menus to Configuration?

Comment (MiquelRForgeFlow) [reply account_move_exception/views/account_view.xml]: Done

**Review by JordiMForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Generated README is incorrect.

**Comment by AaronHForgeFlow** [comment]
/ocabot merge nobump

Thank you @JordiMForgeFlow !

**Pull request merged**